### DISCUSSION TITLE
Questions about passes updates

### DISCUSSION THREAD

**Discussion by tomm3w** [alexandercerutti/passkit-generator#174] [Q&A]
I know that information can be updated on the front of the pass if you have "Automatic Updates" turned on.  Two questions, first, if it is turned on, is there a setting for speeding up the data transfer to the pass to update current miles, or value if creating a gift card, and second, Is there a way to "auto updates" if the window is not showing that option but it can be set up in the backend to lock user out from closing.  

Thanks
Tom

**Comment by alexandercerutti** [comment]
Hi @tomm3w, thanks for using passkit-generator (if you are using it).

I'm not aware of any mechanism that does what you want to achieve.

From what I know, it all depends on the server that sends the pass after it sends the notification and the user opens the updated pass.

I don't know which problems are you facing. Furthermore, I never tried to set up a server that updated the pass.

For the second question, which window are you talking about?

**Reply by alexandercerutti** [reply]
Uhm for the second question, maybe I understand what you are referring to.

I think it might depend on the availability of the fields `webServiceUrl` and `authenticationToken` you set in pass.json.

I tried to compare two passes, one that shows notification settings in Apple Wallet, and one that doesn't. The one that supports updates has the two fields. The other doesn't have them.

So, I guess the answer is no because Apple Wallet doesn't have enough information on how to auto-update them. There is a specific protocol for that, and that involves the usage of `webServiceUrl`  and `authenticationToken`.

<img src="https://github.com/alexandercerutti/passkit-generator/assets/5442545/fcaf2c46-8c3b-406c-a794-4b9bc47777ef" width="250" />

<img src="https://github.com/alexandercerutti/passkit-generator/assets/5442545/4c7af988-4f09-4796-9a14-480ea9784839" width="250" />

**Comment by tomm3w** [comment]
Thanks for your response.  My airline updates my pass with changes and a hotel priority card updated my points.   I didn’t know if what I was asking for was common knowledge Thanks for you responseSent from my iPhoneOn Oct 24, 2023, at 11:40 AM, Alexander Cerutti ***@***.***> wrote:﻿
Uhm for the second question, maybe I understand what you are referring to.
I think it might depend on the availability of the fields webServiceUrl and authenticationToken you set in pass.json.
I tried to compare two passes, one that shows "Notification" settings in Apple Wallet, and one that doesn't. The one that supports updates has two fields. The other doesn't have them.
So, I guess the answer is no because Apple Wallet doesn't have enough information on how to auto-update them. There is a specific protocol for that, and that involves the usage of webServiceUrl  and authenticationToken.

**Reply by alexandercerutti** [reply]
Consider that many things about Apple Wallet are known only to Apple. Sometimes they are not even documented and must be guessed.

Anyway, you are welcome!

If you use passkit-generator, if you could leave a ⭐ on the project, that would be awesome.
